src/test/java/org/apache/commons/crypto/benchmark/CryptoStreamBenchmark.java

```diff
@@ -0,0 +1,211 @@
+        .getCipherClazzName() + " in transformation " + streamOption
+        .getTransformation();
+  }
+
+  OutputStream getCryptoOutputStream() throws IOException {
+    if (cryptoOutputStream == null) {
+      Random r = new Random();
+      byte[] iv = new byte[16];
+      byte[] key = new byte[16];
+      r.nextBytes(iv);
+      r.nextBytes(key);
+
+      Properties props = new Properties();
+      props.put(ConfigurationKeys.CHIMERA_CRYPTO_CIPHER_CLASSES_KEY,
+          streamOption.getCipherClazzName());
+      props.put(ConfigurationKeys.CHIMERA_CRYPTO_STREAM_BUFFER_SIZE_KEY,
+          streamOption.getBufferSize());
+      cryptoOutputStream = new CryptoOutputStream(
+          streamOption.getTransformation(), props, outputStream, key, iv);
+    }
+    return cryptoOutputStream;
+  }
+
+  protected InputStream getCryptoInputStream() throws IOException {
+    if (cryptoInputStream == null) {
+      Random r = new Random();
+      byte[] iv = new byte[16];
+      byte[] key = new byte[16];
+      r.nextBytes(iv);
+      r.nextBytes(key);
+
+      Properties props = new Properties();
+      props.put(ConfigurationKeys.CHIMERA_CRYPTO_CIPHER_CLASSES_KEY,
+          streamOption.getCipherClazzName());
+      props.put(ConfigurationKeys.CHIMERA_CRYPTO_STREAM_BUFFER_SIZE_KEY,
+          streamOption.getBufferSize());
+      cryptoInputStream = new CryptoInputStream(
+          streamOption.getTransformation(), props, inputStream, key, iv);
+    }
+    return cryptoInputStream;
+  }
+
+  protected byte[] prepareData(int size) {
+    byte[] data = new byte[size];
+    Random r = new Random();
+    r.nextBytes(data);
+    return data;
+  }
+
+  public void testEncryption(int iterations) {
+    try {
+      OutputStream cryptoOutputStream = getCryptoOutputStream();
+
+      System.out.println("warming up");
+      // warm up
+      for (int i = 0; i < benchmarkOption.warmupIterations; i++) {
+        doWriteOperation(outputStream, cryptoOutputStream, inputData);
+      }
+
+      System.out.println("warming up complete.");
+
+      long begin = Calendar.getInstance(TimeZone.getTimeZone("UTC"))
+          .getTimeInMillis();
+      for (int i = 0; i < iterations; i++) {
+        doWriteOperation(outputStream, cryptoOutputStream, inputData);
+      }
+
+      long end = Calendar.getInstance(TimeZone.getTimeZone("UTC"))
+          .getTimeInMillis();
+      printResult("===encryption",
+          1000.0 * benchmarkOption.dataSize * iterations / ((end - begin) *
+              1024.0 * 1024.0));
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+  }
+
+  public void testDecryption(int iterations) {
+    try {
+      InputStream cryptoInputStream = getCryptoInputStream();
+
+      byte[] outputData = new byte[benchmarkOption.dataSize];
+
+      System.out.println("Warming up.");
+      // warm up
+      for (int i = 0; i < benchmarkOption.warmupIterations; i++) {
+        doReadOperation(inputStream, cryptoInputStream, outputData);
```

Comment:
We should use the right data for decryption, because for some modes, decryption the random data will throw exceptions.